requestor-tutorials/vm-runtime/golem-certificates.md

```diff
@@ -0,0 +1,80 @@
+# Golem-certificates
+
+## Installing cli tool
+
+Prerequisites:
+
+- [Rust toolchain](https://rustup.rs/)
+- Git
+
+```bash
+git clone https://github.com/golemfactory/golem-certificate
+cargo install --path golem-certificate/cli
+```
+
+After that, in other terminal enter UI tool by typing:
+
+```bash
+golem-certificate-cli ui
+```
+
+## Generating key-pair
+
+Create new key-pair in `Create key pair` option in main menu and name files appropriately.
+
+## Generating root certificate
+
+Prerequisites:
+
+- key-pair for your root certificate
+
+Navigate to `Create certificate` option.
+After that, put your propertiies as needed (Key usage, Validity period, Permissions and subject information) and enter your generated **public** key.
+
+Now you are able to sign your certificate.
+Enter your **private** key to "Signing key" field and set "Signing certificate" as `Self-signed`.
+
+## Generating intermediate certificate
+
+Prerequisites:
+
+- key-pair for your certificate certificate
+- root certificate entity which will sign your certificate template
+
+Navigate to `Create certificate` option.
+After that, put your propertiies as needed (Key usage, Validity period, Permissions and subject information) and enter your generated **public** key.
+
+Then you should save your certificate as template, and send it to root certificate entity which will sign it for you.
+
+## Signing certificate
+
+Prerequisites:
+
+- key-pair for your certificate which will sign other certificate
+- your certificate with appropriate `key-usage`
+- template of certificate which you want to sign
+
+Navigate to `Create certificate` option and load from desired template.
```

Comment:
Maybe add a line to check that the template does not request more permissions / key usage / validity period than the signing certificate has. or at least mention that during the signing process there will be an error if it is the case.